magnum-zio/src/main/scala/com/augustnagro/magnum/magzio/Transactor.scala

```diff
@@ -1,36 +1,32 @@
 package com.augustnagro.magnum.magzio
 
-import com.augustnagro.magnum.{DbCon, DbTx, SqlException, SqlLogger}
+import com.augustnagro.magnum.{
+  DbCon,
+  DbTx,
+  SqlException,
+  SqlLogger,
+  TransactorOps
+}
 import zio.{Semaphore, Task, Trace, UIO, ULayer, ZIO, ZLayer}
 
 import java.sql.Connection
 import javax.sql.DataSource
 import scala.util.control.NonFatal
 
-class Transactor private (
-    dataSource: DataSource,
-    sqlLogger: SqlLogger,
-    connectionConfig: Connection => Unit,
-    semaphore: Option[Semaphore]
-):
+final case class Transactor(
+    private val dataSource: DataSource,
+    private val sqlLogger: SqlLogger,
+    private val connectionConfig: Connection => Unit,
+    private val semaphore: Option[Semaphore]
+) extends TransactorOps[Task]:
 
   def withSqlLogger(sqlLogger: SqlLogger): Transactor =
-    new Transactor(
-      dataSource,
-      sqlLogger,
-      connectionConfig,
-      semaphore
-    )
+    copy(sqlLogger = sqlLogger)
 
-  def withConnectionConfig(connectionConfig: Connection => Unit): Transactor =
-    new Transactor(
-      dataSource,
-      sqlLogger,
-      connectionConfig,
-      semaphore
-    )
+  def withConnectionConfig(connectionConfig: Connection => Unit) =
+    copy(connectionConfig = connectionConfig)
 
-  def connect[A](f: DbCon ?=> A)(using Trace): Task[A] =
```

Comment:
I think the need to remove the `(using Trace)` parameter here shows the limit of the common trait idea. I don't really agree with the idea. I think leaving each integration free of bringing the best integration to its environment without constraints is the best approach. 

Wy would we need this trait? It's not like, as a Magnum user, you'd use it anyway to abstract over any Transator implementation.

Reply:
As a library author you would use it to generate a test kit - such that the tests are written to the trait and not to the impls - that makes it easier to plug the impls into the core tests without rewriting them.
To the library user they need not even know about the trait - notice the library still works just as it did before - a user need not care. But library maintainers often just want one set of core tests.

Reply:
When I look at most ZIO services - I don't see (using trace) anywhere on service methods - I am not sure we need it at all. If we find we do - I can think of a couple of ways to support it.